extensions/histogram/src/main/java/io/druid/query/aggregation/histogram/ApproximateHistogramAggregatorFactory.java

```diff
@@ -115,6 +116,36 @@ public AggregatorFactory getCombiningFactory()
   }
 
   @Override
+  public AggregatorFactory getMergingFactory(AggregatorFactory other) throws AggregatorFactoryNotMergeableException
+  {
+    if (other.getName().equals(this.getName()) && other instanceof ApproximateHistogramAggregatorFactory) {
+      ApproximateHistogramAggregatorFactory castedOther = (ApproximateHistogramAggregatorFactory) other;
+
+      if (Float.compare(castedOther.lowerLimit, lowerLimit) == 0 &&
+          numBuckets != castedOther.numBuckets &&
+          resolution != castedOther.resolution &&
+          Float.compare(castedOther.upperLimit, upperLimit) != 0) {
+        return getCombiningFactory();
+      } else {
+        throw new AggregatorFactoryNotMergeableException(
+            "can't merge approx histogram aggregator factories [%s] and [%s]",
+            this,
+            other
+        );
+      }
+
+    } else {
+      throw new AggregatorFactoryNotMergeableException(
```

Comment:
Possibly worth having a constructor for the exception that takes two AggregatorFactories and just does this message.